Libraries/Text/RCTTextAttributes.m

```diff
@@ -225,7 +229,7 @@ - (CGFloat)effectiveFontSizeMultiplier
 
 - (RCTUIColor *)effectiveForegroundColor // TODO(OSS Candidate ISS#2710739)
 {
-  RCTUIColor *effectiveForegroundColor = _foregroundColor ?: [RCTUIColor blackColor]; // TODO(OSS Candidate ISS#2710739)
+  RCTUIColor *effectiveForegroundColor = _foregroundColor ?: [RCTUIColor labelColor]; // TODO(OSS Candidate ISS#2710739)
```

Comment:
Should I use the macOS target conditional there since [upstream](https://github.com/facebook/react-native/blob/master/ReactCommon/fabric/textlayoutmanager/platform/ios/RCTAttributedTextUtils.mm#L87) is still using `blackColor` there?

Reply:
Actually i believe we need to keep the default foreground text color as black for both platforms.   The reason is that in an NSAttributedString, when a foreground color is not specified explicitly, the default color *is* black.   This is true on iOS and macOS.   Changing this could break some logic in native code.

Its still possible to set the foreground color to 'labelColor' for either platform, but it should be done on the JavaScript side.